lib/internal/Magento/Framework/App/FrontController.php

```diff
@@ -156,15 +185,11 @@ private function processRequest(
         //Validating a request only once.
         if (!$this->validatedRequest) {
             try {
-                $this->requestValidator->validate(
-                    $request,
-                    $actionInstance
-                );
+                $this->requestValidator->validate($request, $actionInstance);
```

Comment:
$request parameter should be of type RequestInterface sonce it passed from dispatch() method. Please change parameter type, hinting and annotation

Reply:
I have question here 
In my opinion we should 
CC: @kandy @paliarush

Reply:
![Screenshot from 2021-02-09 11-06-20](https://user-images.githubusercontent.com/1908873/107314378-10edfb00-6ac7-11eb-880a-73e544398c6a.png)
Here is snippets demo

Reply:
@zakdma 
> $request parameter should be of type RequestInterface sonce it passed from dispatch() method. Please change parameter type, hinting and annotation
Since the method `getFullActionName` is called on the request, the dependency is actually on the `HttpRequest`, not the `RequestInterface`.
Depending on `RequestInterface` means an interface contract violation.
The options are to either add the method to the `RequestInterface` (backward incompatible change), or to depend on the `HttpRequest` (which is simply making the implicit explicit).